### ISSUE TITLE
[turborepo] Turbo prune removes patchedDependencies

### ISSUE THREAD

**Issue by marekvospel** [vercel/turbo#2788]
### What version of Turborepo are you using?

1.6.3

### What package manager are you using / does the bug impact?

pnpm

### What operating system are you using?

Linux

### Describe the Bug

When running `turbo prune`, `pnpm.patchedDependencies` is set to `{}` in the out directory.

### Expected Behavior

`patchedDependencies` should stay as they were before running `tubro prune`, or an option to keep some / all patched dependencies should be added. 

### To Reproduce

1. Create a turborepo with a package and install a dependency to it
2. Patch the dependency using `pnpm patch` and `pnpm commit-patch`
3. Run `turbo prune --scope=package` and observe `out/package.json`

In the reproduction repo I've already done this, so both the input and output can be seen in the repo.

### Reproduction Repo

https://github.com/marekvospel/turborepo-prune-removes-patched

**Comment by gajus** [comment]
Same issue

**Comment by gajus** [comment]
Not ideal, but I have a workaround.

```Dockerfile
# Node version must match with .nvmrc file
FROM node:18-bullseye AS pruner

# Workaround for https://github.com/vercel/turbo/issues/2788
RUN apt update && apt install jq moreutils -y && \
  npm install turbo@^1 --global

WORKDIR /srv

COPY . .

RUN turbo prune --scope=contra-web-app --docker

RUN export pd=$(jq .pnpm.patchedDependencies package.json) && jq --argjson pd "${pd}" '.pnpm.patchedDependencies = $pd' out/json/package.json | sponge out/json/package.json

COPY patches out/json/patches

FROM node:18-bullseye AS installer

# ...
```

You also need to set `allowNonAppliedPatches`.

**Comment by marekvospel** [comment]
Closing, as this was fixed in 1.7.1

**Comment by vitorcamachoo** [comment]
This is still happening on **turbo@2.2.3** 😞
